### ISSUE TITLE
Support for metrics and tracing in Javalin web framework

### ISSUE THREAD

**Issue by davidcurrie** [open-telemetry/opentelemetry-java-instrumentation#6115]
**Is your feature request related to a problem? Please describe.**
Support for Open Telemetry metrics and tracing in the [Javalin](https://javalin.io/) web framework.

**Describe the solution you'd like**
To have automatic support for Open Telemetry metrics and tracing in Javalin.

**Describe alternatives you've considered**
No alternatives were considered.

**Additional context**
See discussion in https://github.com/tipsy/javalin/issues/523

**Comment by trask** [comment]
hi @davidcurrie! if you are interested in working on this, check out [writing-instrumentation.md](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/writing-instrumentation.md), [writing-instrumentation-module.md](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/writing-instrumentation-module.md), and [using-instrumenter-api.md](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/using-instrumenter-api.md)

**Comment by PhilKes** [comment]
I was taking a look at this and even though there is no explicit support for javalin, metrics and spans are already created with the otel-javaagent. From what I can see the current version `javalin-2.0.3` uses `jetty-11.0`, which is already supported by OpenTelemetry.
I set up an example javalin project with a simple REST Api and using the current `opentelemetry-javaagent.jar`, and both Spans for incoming HTTP Request as well as `http_server` metrics were created.
So is this already resolved?

**Comment by davidcurrie** [comment]
Sorry - somehow missed the question back in Feb. From what I remember, the issue with Javalin was that it's implemented as a single servlet hanging off `/*` so that's the path that was showing up on all the Jetty-based metrics. Unfortunately, we've been unable to continue with OpenTelemetry due to OSGi issues so I can't comment further and understand if this issue just gets closed out.

**Comment by thisisphan** [comment]
> From what I remember, the issue with Javalin was that it's implemented as a single servlet hanging off /* so that's the path that was showing up on all the Jetty-based metrics

Yeah, getting official otel support to finer metrics would be great.
